CARL/source/Recognizer.cpp

```diff
@@ -62,26 +58,37 @@ namespace carl::action
                 }
                 else
                 {
-                    double t = (timestamp - original.getStartTimestamp()) /
-                        (original.getEndTimestamp() - original.getStartTimestamp());
```

Comment:
This is the kind of bug that makes me question whether the bug knows something I no longer remember. It's unsurprising that this wouldn't have caused much of a problem since it would have simply done the wrong interpolation between two sequential samples, which should vary timing by only a very tiny amount. However, with the introduction of `std::ceil` mentioned above, `timestamp` can sometimes have values larger than `endTimestamp`, which could theoretically extrapolate samples that were never implied by the original recording.